src/lib/hydrate.ts

```diff
@@ -209,25 +220,20 @@ export const hydrate =
               // need to stop hydrating this subtree? Clear it? Add tests.
               throw new Error('unimplemented');
             }
+          } else if (isIterable(value)) {
+            // currentNodePart.value will contain an array of NodeParts
+            stack.push({
+              part: currentNodePart,
+              type: 'iterable',
+              value,
+              iterator: value[Symbol.iterator](),
+              done: false,
+            });
+            currentNodePart.value = [];
           } else {
-            if (typeof value === 'string') {
-              // TODO: implement the rest of the NodePart commit() cascade. Can
-              // we reuse the code?
-              stack.push({part: currentNodePart, type: 'leaf'});
-              currentNodePart.value = value;
-            } else if (isIterable(value)) {
-              // currentNodePart.value will contain an array of NodeParts
-              stack.push({
-                part: currentNodePart,
-                type: 'iterable',
-                value,
-                iterator: value[Symbol.iterator](),
-                done: false,
-              });
-              currentNodePart.value = [];
-            } else {
-              stack.push({part: currentNodePart, type: 'leaf'});
-            }
+            // Objects, Nodes, Functions, etc...
```

Comment:
In `NodePart.commit`, after iterable, there's the `value === nothing` case. Does that need to be handled here?

Reply:
In hydration we're only initializing the part's `value` (not doing anything with the actual rendered content), so no, `nothing` is treated like all the other fallback values, in that its `part.value` is just the value (`nothing`). It would be extra code to separate that out into another clause to match the NodePart.commit code, so I'm just noting that nothing is handled in the comment.

I did go back and look and notice that the fallback does initialize `part.value` to empty string for `null` and `undefined`; I don't think it really matters since fallback values aren't dirty checked anyway, but went ahead and updated the hydrate code to match.